transmogrifier/helpers.py

```diff
@@ -83,6 +85,90 @@ def parse_xml_records(
             element.clear()
 
 
+def parse_date_from_string(
+    date_string: str,
+) -> Optional[datetime]:
+    """
+    Transform a date string into a datetime object according to one of the configured
+    OpenSearch date formats.
+
+    Args:
+        date_string: A date string.
+        source_record_id: The ID of the record being transformed.
```

Comment:
This arg should be removed from the docstring. I would probably also note here that this returns `None` if the date string can't be parsed.